### PR TITLE
Feature disable current user from modifying their own groups

### PR DESCRIPTION
**Pull request by flamerohr**
Fixes https://github.com/silverstripe/silverstripe-framework/issues/7304
Fixes https://github.com/silverstripe/silverstripe-cms/issues/730

### CODE REVIEW

**Comment by tractorcow** [comment]
There was a commit (never merged) that did something similar.

https://github.com/silverstripe/silverstripe-framework/pull/2782/commits/eaf966523713be8e558a41768e9039e482075774

Except that it was a warning, not an error, and it only applied to admins.

There are times it is necessary for a user to remove their own group, so I'm not sure this feature is strictly correct. Also it should probably only apply to admin users / groups.

**Comment by tractorcow** [comment]
My other concern is the switch from a combo box to a full gridfield. It makes the UX more complicated to use.

Reassigning back to @chillu to provide some product owner oversight.

File: src/Forms/GridField/GridFieldGroupDeleteAction.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        $currentUser = Security::getCurrentUser();
+
+        if ($actionName == 'unlinkrelation' && (
+            $this->userID === $currentUser->ID || (
+                $gridField->getList()->first() instanceof Member &&
+                $currentUser->ID === $arguments['RecordID']
+            )
+        )) {
+            throw new ValidationException(
+                _t(self::class . '.DeleteSelfFailure', 'Cannot remove yourself from groups')
```
Comment (dhensby) [inline src/Forms/GridField/GridFieldGroupDeleteAction.php]: my understanding is the text collector cant handle `::class` and we have to use `__CLASS__` or strings

Comment (flamerohr) [reply src/Forms/GridField/GridFieldGroupDeleteAction.php]: yeah true, I forgot about `::class` not being supported

I'll include that in my rework

**Comment by flamerohr** [comment]
Have updated pull request and added behat tests

**Comment by tractorcow** [comment]
Pushed up some fixes and code cleanup.

**Pull request merged**